common/src/main/java/io/homeassistant/companion/android/common/data/integration/impl/IntegrationRepositoryImpl.kt

```diff
@@ -167,7 +167,7 @@ class IntegrationRepositoryImpl @Inject constructor(
                         "render_template",
                         mapOf("template" to Template(template, variables))
                     )
-                ).getValue("template")
+                ).getValue("template") ?: throw NullPointerException("Null returned in template")
```

Comment:
I ran into an error I didn't see before. The app would crash when HA returns null for the template. According to AS this can never be null, but still this change was required to stop the app from crashing

Reply:
Do you have an example of a template that returns `null`?

If null is a valid result from the server, wouldn't it make more sense to change the return type to `String?`?

Reply:
Yes I do. I store my content in a binary sensor attribute for easier editing. So in my tile settings I just use something like

`{{ state_attr('binary_sensor.watch_template_test', 'template_text') }}`

I received the null when I started to add the `_test` suffix. So I got a non-existing entity id: `binary_sensor.watch_template_`

So it seems to return null when the template is invalid and/or when using an invalid entity id. Not sure on the exact conditions. I'm also fine with returning `String?` and handling the null elsewhere. However, the `getValue` seems to return a non-nullable `String`.

Reply:
It looks like the server actually returns `null` while indicating it was succesful, so I'd keep that and not throw an error. The developer tools in the frontend also show 'null' instead of an error.

```
2022-07-01 13:35:19.210 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: --> POST https://[domain]/api/webhook/[webhook id]
2022-07-01 13:35:19.210 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Content-Type: application/json; charset=UTF-8
2022-07-01 13:35:19.210 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Content-Length: 148
2022-07-01 13:35:19.210 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: {"type":"render_template","data":{"template":{"template":"{{ state_attr('binary_sensor.watch_', 'template_text') }} ","variables":{}}}}
2022-07-01 13:35:19.210 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: --> END POST (148-byte body)
2022-07-01 13:35:19.242 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: <-- 200 OK https://[domain]/api/webhook/[webhook id] (31ms)
2022-07-01 13:35:19.242 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Content-Type: application/json; charset=utf-8
2022-07-01 13:35:19.242 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Content-Length: 18
2022-07-01 13:35:19.242 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Date: Fri, 01 Jul 2022 11:35:19 GMT
2022-07-01 13:35:19.242 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: Server: Python/3.9 aiohttp/3.8.1
2022-07-01 13:35:19.243 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: {"template": null}
2022-07-01 13:35:19.243 15428-15566/io.homeassistant.companion.android.debug I/okhttp.OkHttpClient: <-- END HTTP (18-byte body)
```

> However, the getValue seems to return a non-nullable `String`

You can use `getOrDefault("template", null)`.

Reply:
So `getValue` either returns the value or an exception, while `get` will return the value or null if it doesn't exist. So it seems that switching to `get` (or `getOrDefault`) will change the behavior, in that we won't get an exception if the return value didn't contain the key, but just get null instead.

I think we will then still get an exception when we cannot connect to the server, so we can show a different message for when catching an exception versus receiving `null`. Does that make sense to implement?

Reply:
But the key does exist (see the server response in my previous comment), so does it correctly recognize it as `null` or throw an exception because you're trying to cast null to a non-nullable String?

> I think we will then still get an exception when we cannot connect to the server, so we can show a different message for when catching an exception versus receiving `null`. Does that make sense to implement?

Yes, you're correct and good idea!

Reply:
Btw, seem related to #2157